modules/ROOT/pages/provisioning-hetzner.adoc

```diff
@@ -0,0 +1,180 @@
+
+You also need to have access to a Hetzner account.
+The examples below use the https://github.com/hetznercloud/cli[hcloud] command-line tool, the https://github.com/apricote/hcloud-upload-image[hcloud-upload-image] tool and https://stedolan.github.io/jq/[jq] as a command-line JSON processor.
+
+== Creating a snapshot
+
+Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+
+. Once you have picked the relevant stream, download the latest RAW image from the https://fedoraproject.org/coreos/download/?stream=stable#baremetal[download page] or with podman (see https://coreos.github.io/coreos-installer/cmd/download/[documentation] for options):
++
+[source, bash]
+----
+arch="x86_64" # or aarch64
+podman run --security-opt label=disable --pull=always --rm -v .:/data -w /data \
+    quay.io/coreos/coreos-installer:release download -s stable -p metal -f raw -a "${arch}"
+----
++
+Note this is just using `coreos-installer` as a tool to download the RAW disk image.
++
+NOTE: Both x86_64 and aarch64 architectures are supported on Hetzner.
+
+. Copy paste the following Bash script into a file name `convert-image.sh`:
++
+.QEMU to Hetzner image conversion script
+[source, bash]
+----
+#!/bin/bash
+
+set -euo pipefail
+
+if [[ ${#} -ne 3 ]]; then
+    echo "Usage: <source image> <dest image> <platform>"
+    echo ""
+    echo "Example:"
+    echo "./$(basename "${0}") fedora-coreos-40.20240616.3.0-{metal,heztner}.x86_64.raw heztner"
+    exit 1
+fi
+
+source="${1}"
+dest="${2}"
+platform="${3}"
+
+if [[ ! -f "${source}" ]]; then
+    echo "Source image ${source} does not exists"
+    exit 1
+fi
+
+if [[ -f "${dest}" ]]; then
+    echo "Destination image ${dest} already exists"
+    exit 1
+fi
+
+if [[ -z "$(command -v guestfish)" ]]; then
+    echo "Could not find 'guestfish' command"
+    exit 1
+fi
+
+cp --reflink=auto "${source}" "${dest}"
+guestfish -a "${dest}" <<EOF
+run
+mount /dev/sda3 /
+download /loader/entries/ostree-1.conf tmp.loader.entries.ostree-1.conf
+<! sed -i "s/ignition.platform.id=metal/ignition.platform.id=${platform}/" tmp.loader.entries.ostree-1.conf
+upload tmp.loader.entries.ostree-1.conf /loader/entries/ostree-1.conf
+EOF
+
+rm -v ./tmp.loader.entries.ostree-1.conf
+
+echo "Done"
+----
++
+. Make sure that you have `guestfish` installed on your system and convert the metal image to an Hetzner one:
++
+[source, bash, subs="attributes"]
+----
+source_image"fedora-coreos-{stable-version}-metal.x86_64.raw"
+image_name="fedora-coreos-{stable-version}-hetzner.x86_64.raw"
+./convert-image.sh "${source_image}" "${image_name}" hetzner
+----
++
+. Compress the image using `bzip2`:
++
+[source, bash]
+----
+bzip2 --compress --fast "${image_name}"
+----
++
+. Upload the image somewhere publicly available over HTTPS and note down the URL:
```

Comment:
`hcloud-upload-image` also supports using the local file directly since 0.2.0. I think that would be easier for new users in comparison to uploading the file to some public URL. In the end both are limited by the local upload speed. 

Do you see any downside of that?

Reply:
Ah, thanks! I had missed that. Indeed, this should make it much easier.

Reply:
Fixed.